app/src/main/java/org/mozilla/fenix/composables/StaggeredHorizontalGrid.kt

```diff
@@ -0,0 +1,137 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.composables
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.Placeable
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Displays a list of items as a staggered horizontal grid placing them on ltr rows and continuing
+ * on as many below rows as needed to place all items.
+ *
+ * In an effort to best utilize the available row space this can mix the items such that narrower ones
+ * are placed on the same row as wider ones if the otherwise next item doesn't fit.
+ *
+ * @param modifier [Modifier] to be applied to the layout.
+ * @param horizontalItemsSpacing minimum horizontal space between items. Does not add spacing to layout bounds.
+ * @param verticalItemsSpacing vertical space between items
+ * @param arrangement how the items will be horizontally aligned and spaced.
+ * @param content the children composables to be laid out.
+ */
+@Composable
+fun StaggeredHorizontalGrid(
```

Comment:
I am wondering if you also saw this https://google.github.io/accompanist/flowlayout/? I imagine at some point we don't want to maintain our own flow layout.

Reply:
`flowlayout` has two important issues right now for this feature at least:
- no RTL support while `StaggeredHorizontalGrid` correctly handles RTL also.
- lays out all items in the exact order. This can be ok for most cases but it can also lead to situations in which although there is enough space left on one row if the next immediate items does not fit there then that space is left empty. 
So instead of the grid being neatly packed, more closely following the spacing desired the spacing will be almost random.
For this feature the items are rather random while they all should be aligned to start with the same spacing for a clean design, scenario for which `StaggeredHorizontalGrid` suited.

In the future we could use `flowlayout` or tweak this whenever we need withought depending on external functionality/behavior we have little control over.